src/structurizr/api/structurizr_client.py

```diff
@@ -109,14 +114,36 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        """Exit a context by unlocking the corresponding remote workspace."""
+        """Exit a context by unlocking the corresponding remote workspace.
+
+        Note: this method is deprecated in favour of lock(), and will be removed in a
+        future relesae.
+        """
         is_successful = self.unlock_workspace()
         self._client.close()
         if exc_type is None and not is_successful:
             raise StructurizrClientException(
                 f"Failed to unlock the Structurizr workspace {self.workspace_id}."
             )
 
+    @contextmanager
+    def lock(self):
+        """Provide a context manager for locking and unlocking a workspace."""
+        is_successful = self.lock_workspace()
+        if not is_successful:
+            raise StructurizrClientException(
+                f"Failed to lock the Structurizr workspace {self.workspace_id}."
+            )
+        try:
+            yield None
```

Comment:
I think for consistency we should yield `self` here even though this value will most often be ignored.

```suggestion
            yield self
```